### ISSUE TITLE
Inconsistency in results

### ISSUE THREAD

**Issue by divamgupta** [bluemonk482/tdlstm#2]
The TD LSTM gives very inconsistent results everytime we train the model. What is the protocol used for reporting the results. Do we run it multiple times and report the best test accuracy?

**Comment by bwang482** [comment]
Hi @divamgupta thanks for the interest !

I have experienced the same inconsistency in performance, thus why I didn't report the result of TDLSTM on our multi-target election corpus in our [EACL 2017 paper](https://aclanthology.info/pdf/E/E17/E17-1046.pdf). As for the single-target corpus, we included the result that was reported in Tang's [original TDLSTM paper](https://arxiv.org/pdf/1512.01100.pdf) (rather than what we obtained ourselves using out reimplementation code). 

The same issue has been reported by the others, not only for TDLSTM but also other DL networks, possibly due to the lack of training data: [Link 1](https://goo.gl/ApD5ku) and [Link 2](https://github.com/LeonCrashCode/AttentionTargetSentiment/issues/3).

Do you get different results using the same initialisation weights and parameters, and the same random seeds?

**Comment by divamgupta** [comment]
On different machines, we get different results even with the same initialization weights and parameters, and the same random seeds. So this makes the paper non-reproducable.
Is it a valid protocol to run the model multiple times and report the best results?

**Comment by bwang482** [comment]
Hmm, this is what the others reported as well. I am not the author of the TDLSTM or TCLSTM paper, and in my paper I only included their reported result. Have you tried this implementation version of TDLSTM: https://github.com/scaufengyang/TD-LSTM? Please let me know if you get more consistent result using this code, so I know whether or not the issue lies in my implementation.

**Comment by bwang482** [comment]
@divamgupta any updates? Have you managed to reproduce their results?

**Comment by divamgupta** [comment]
No
